pkg/oceanbase/operation/backup.go

```diff
@@ -11,3 +11,290 @@ See the Mulan PSL v2 for more details.
 */
 
 package operation
+
+import (
+	"github.com/oceanbase/ob-operator/api/v1alpha1"
+	"github.com/oceanbase/ob-operator/pkg/oceanbase/const/sql"
+	"github.com/oceanbase/ob-operator/pkg/oceanbase/model"
+	"github.com/pkg/errors"
+)
+
+func (m *OceanbaseOperationManager) SetLogArchiveDestForTenant(uri string) error {
+	return m.SetParameter("LOG_ARCHIVE_DEST", uri, nil)
+}
+
+func (m *OceanbaseOperationManager) SetLogArchiveConcurrency(concurrency int) error {
+	return m.SetParameter("log_archive_concurrency", concurrency, nil)
+}
+
+func (m *OceanbaseOperationManager) SetDataBackupDestForTenant(uri string) error {
+	return m.SetParameter("DATA_BACKUP_DEST", uri, nil)
+}
+
+func (m *OceanbaseOperationManager) EnableArchiveLogForTenant() error {
+	return m.ExecWithDefaultTimeout(sql.EnableArchiveLog)
+}
+
+func (m *OceanbaseOperationManager) DisableArchiveLogForTenant() error {
+	return m.ExecWithDefaultTimeout(sql.DisableArchiveLog)
+}
+
+func (m *OceanbaseOperationManager) SetBackupPassword(password string) error {
+	return m.ExecWithDefaultTimeout(sql.SetBackupPassword, password)
+}
+
+func (m *OceanbaseOperationManager) CreateBackupFull() error {
+	return m.ExecWithDefaultTimeout(sql.CreateBackupFull)
+}
+
+func (m *OceanbaseOperationManager) CreateBackupIncr() error {
+	return m.ExecWithDefaultTimeout(sql.CreateBackupIncr)
+}
+
+func (m *OceanbaseOperationManager) CreateAndReturnBackupJob(jobType v1alpha1.BackupJobType) (*model.OBBackupJob, error) {
+	var err error
+	if jobType == v1alpha1.BackupJobTypeFull {
+		err = m.ExecWithDefaultTimeout(sql.CreateBackupFull)
+	} else if jobType == v1alpha1.BackupJobTypeIncr {
+		err = m.ExecWithDefaultTimeout(sql.CreateBackupIncr)
+	} else {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return m.QueryLatestBackupJobOfType(jobType)
```

Comment:
is it guaranteed to generate a new job instance in db that can be queried out once a job is created
if there is a delay, then the queried object maybe an older job instance

Reply:
Hundreds of called operations show that it's almost guaranteed. If it return nil at first, Reconciler will fetch the job record from database view soon.